core/modules/field/field.crud.inc

```diff
@@ -513,6 +513,16 @@ function field_create_instance($instance) {
     throw new FieldException($message);
   }
 
+  // Check to ensure that the field instance does not have a reserved word for
+  // this entity type.
+  $entity_info = entity_get_info($instance['entity_type']);
```

Comment:
I'm wondering if this check should instead be done in `field_validate_instance()`, to keep the same pattern as in the creation of new fields (a similar check is performed in `field_validate_field()`.

Reply:
OK. By changing the update one to a `backdrop_set_message()` of type `error` it avoids the full screen error which is also breaking tests.